lib/libc/picolibc/write.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright © 2021, Keith Packard <keithp@keithp.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/libc-hooks.h>
+
+/* Don't replace the version supplied by fdtable */
+#if !defined(CONFIG_FDTABLE) || !defined(CONFIG_POSIX_API)
```

Comment:
At some point, because `CONFIG_POSIX_API` will be deprecated.

Would it make sense to have Picolibc select `CONFIG_POSIX_DEVICE_IO` and  `CONFIG_TC_PROVIDES_POSIX_DEVICE_IO` since it requires `write()` and maybe other symbols? They would still be provided by picolibc.

There would be no need for the ifdefs at that point.

Just a comment though, no changes needed currently.